dnf/cli/commands/module.py

```diff
@@ -0,0 +1,238 @@
+
+        aliases = ('list',)
+
+        def configure(self):
+            demands = self.cli.demands
+            demands.available_repos = True
+            demands.sack_activation = True
+
+        def run_on_module(self):
+            mods = self.base.repo_module_dict
+
+            if self.opts.all:
+                print(mods.get_brief_description_all(self.opts.module_nsvp))
+            elif self.opts.enabled:
+                print(mods.get_brief_description_enabled(self.opts.module_nsvp))
+            elif self.opts.disabled:
+                print(mods.get_brief_description_disabled(self.opts.module_nsvp))
+            elif self.opts.installed:
+                print(mods.get_brief_description_installed(self.opts.module_nsvp))
+            else:
+                print(mods.get_brief_description_latest(self.opts.module_nsvp))
+            return 0
+
+    class InfoSubCommand(SubCommand):
+
+        aliases = ('info',)
+
+        def configure(self):
+            demands = self.cli.demands
+            demands.available_repos = True
+            demands.sack_activation = True
+
+        def run_on_module(self):
+            for spec in self.opts.module_nsvp:
+                if self.opts.verbose:
+                    logger.info(self.base.repo_module_dict.get_full_info(spec))
+                else:
+                    logger.info(self.base.repo_module_dict.get_info(spec))
+                print()
+
+    class EnableSubCommand(SubCommand):
+
+        aliases = ('enable',)
+
+        def configure(self):
+            demands = self.cli.demands
+            demands.available_repos = True
+            demands.sack_activation = True
+            demands.root_user = True
+
+        def run_on_module(self):
+            for module_ns in self.opts.module_nsvp:
+                self.base.repo_module_dict.enable(module_ns, self.opts.assumeyes,
+                                                  self.opts.assumeno)
+                logger.info("'{}' is enabled".format(module_ns))
+
+    class DisableSubCommand(SubCommand):
+
+        aliases = ('disable',)
+
+        def configure(self):
+            demands = self.cli.demands
+            demands.available_repos = True
+            demands.sack_activation = True
+            demands.root_user = True
+
+        def run_on_module(self):
+            for module_n in self.opts.module_nsvp:
+                self.base.repo_module_dict.disable(module_n)
+                logger.info("'{}' is disabled".format(module_n))
+
+    class InstallSubCommand(SubCommand):
+
+        aliases = ('install',)
+
+        def configure(self):
+            demands = self.cli.demands
+            demands.available_repos = True
+            demands.sack_activation = True
+            demands.resolving = True
+            demands.root_user = True
+
+        def run_on_module(self):
+            self.base.repo_module_dict.install(self.opts.module_nsvp, self.opts.autoenable)
+
+    class UpdateSubCommand(SubCommand):
```

Comment:
I guess that with this setup, it will be impossible to use general options after sub command. The same problem has repositorypkgs command. The group use single class and no problem with argparser.